app/src/main/java/com/android/streetworkapp/utils/GraphData.kt

```diff
@@ -0,0 +1,152 @@
+package com.android.streetworkapp.utils
+
+import android.annotation.SuppressLint
+import androidx.compose.foundation.Canvas
+import androidx.compose.foundation.gestures.detectTapGestures
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.StrokeCap
+import androidx.compose.ui.input.pointer.pointerInput
+import androidx.compose.ui.platform.LocalDensity
+import androidx.compose.ui.unit.dp
+import com.android.streetworkapp.ui.theme.ColorPalette.INTERACTION_COLOR_DARK
+import kotlin.math.abs
+
+data class GraphData(val x: Float, val y: Float)
+
+class GraphConfiguration(
+    var graphColor: Color = Color.Blue,
+    var axisColor: Color = Color.Black,
+    var xUnitLabel: String = "Time",
+    var yUnitLabel: String = "Reps",
+    var dataPoints: List<GraphData> = emptyList(),
+    var strokeWidth: Float = 4f,
+    var showDashedLines: Boolean = true
+)
+
+@SuppressLint("DefaultLocale")
+@Composable
+fun Graph(modifier: Modifier = Modifier, graphConfiguration: GraphConfiguration) {
```

Comment:
Docs: adding documentation for this function could be useful, since the graph will probably in multiple part of the UI by different members that might be unfamiliar with this function.

(The other elements of the file could also be documented, but I think that this function in particular should be carefully documented for futur utilisation.)